tests/semantic_analyser.cpp

```diff
@@ -1067,6 +1068,61 @@ TEST(semantic_analyser, call_len)
   test("kprobe:f { $x = 0; len($x); }", 1);
 }
 
+TEST(semantic_analyser, call_has_key)
+{
+  test("kprobe:f { @x[1] = 0; if (has_key(@x, 1)) {} }");
+  test("kprobe:f { @x[1, 2] = 0; if (has_key(@x, (3, 4))) {} }");
+  test("kprobe:f { @x[1, (int8)2] = 0; if (has_key(@x, (3, 4))) {} }");
+  test(R"(kprobe:f { @x[1, "hi"] = 0; if (has_key(@x, (2, "bye"))) {} })");
+  test(
+      R"(kprobe:f { @x[1, "hi"] = 0; if (has_key(@x, (2, "longerstr"))) {} })");
+  test(
+      R"(kprobe:f { @x[1, "longerstr"] = 0; if (has_key(@x, (2, "hi"))) {} })");
+  test("kprobe:f { @x[1, 2] = 0; $a = (3, 4); if (has_key(@x, $a)) {} }");
```

Comment:
Maybe also a unit test for passing the key via a map?
```suggestion
  test("kprobe:f { @x[1, 2] = 0; @a = (3, 4); if (has_key(@x, @a)) {} }");
```